src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpointImpl.java

```diff
@@ -0,0 +1,307 @@
+     * @param eventTypes - the event types
+     * @return - a json representation of the event types
+     */
+    private JsonObject getEventTypesJson(List<EventType> eventTypes) {
+        JsonArray eventTypesJson = new JsonArray();
+        for (EventType eventType : eventTypes) {
+            eventTypesJson.add(getEventTypeJson(eventType));
+        }
+        return eventTypesJson.getAsJsonObject();
+    }
+
+    /**
+     * This method returns events as a JsonObject.
+     * @param events - the events
+     * @return - a json representation of the events
+     */
+    private JsonObject getEventsJson(List<Event> events) {
+        JsonArray eventsJson = new JsonArray();
+        for (Event event : events) {
+            List<Attribute> attributes = PersistenceAdapterImpl.getInstance().getAttributes(event.getId());
+            JsonArray jsonAttributes = getAttributesJson(attributes);
+            eventsJson.add(jsonAttributes);
+        }
+        return eventsJson.getAsJsonObject();
+    }
+
+    /**
+     * This method returns attributes as a JsonObject.
+     * @param attributes - the attributes
+     * @return - a json representation of the attributes
+     */
+    private JsonArray getAttributesJson(List<Attribute> attributes) {
+        JsonArray jsonAttributes = new JsonArray();
+        for (Attribute attribute : attributes) {
+            JsonObject jsonAttribute = new JsonObject();
+            TypeAttribute typeAttribute = PersistenceAdapterImpl.getInstance().getTypeAttribute(attribute.getTypeAttributeId());
+            jsonAttribute.addProperty("Name", typeAttribute.getName());
+            jsonAttribute.addProperty("Value", attribute.getValue());
+            jsonAttributes.add(jsonAttribute);
+        }
+        return jsonAttributes;
+    }
+
+    /**
+     * This method returns the id given in request.
+     * @param request the request with entity id
+     * @return id of the entity in request
+     */
+    private long getEntityId(Request request) {
+        return endpointUtil.validateLong(
+                request.params(EntityEndpoint.getEntityIdParameter(false)),
+                (Long input) -> input > 0);
+    }
+
+    /**
+     * This method returns the id given in request.
+     * @param request the request with entity type id
+     * @return id of the entity type in request
+     */
+    private long getEntityTypeId(Request request) {
+        return endpointUtil.validateLong(
+                request.params(EntityEndpoint.getEntityTypeIdParameter(false)),
+                (Long input) -> input > 0);
+    }
+
+    /**
+     * This method returns the start index given in request.
+     * @param request the request with start index
+     * @return start index in request
+     */
+    private int getStartIndex(Request request) {
+        return endpointUtil.validateInteger(
+                request.params(EntityEndpoint.getIndexFromParameter(false)),
+                (Integer input) -> input > 0);

```

Comment:
This should be `(Integer input) -> input >= 0` since `0` is a valid input at this point.